### PR TITLE
Remove paginator and limits on search endpoint

### PR DESCRIPTION
**Pull request by banesullivan**
Simple solution to the limit issue in #354 

This removes the paginator and `limit` argument

cc @matt-bernstein

### CODE REVIEW

File: rgdc/rgdc/rgdc.py [hunk 1]
```diff
@@ -183,7 +183,6 @@ def search(
         resolution: Optional[Tuple[int, int]] = None,
         cloud_cover: Optional[Tuple[float, float]] = None,
         frame_rate: Optional[Tuple[int, int]] = None,
-        limit: Optional[int] = None,
         offset: Optional[int] = None,
```
Comment (zachmullen) [inline rgdc/rgdc/rgdc.py:186]: Limit and offset typically go together. If our intent is to disable all pagination capabilities, we should remove both of them.

**Review by zachmullen — CHANGES_REQUESTED** [review]

File: rgdc/rgdc/rgdc.py (hunk 1 above)
Comment (banesullivan) [reply rgdc/rgdc/rgdc.py:186]: Ah thanks for catching that

**Comment by banesullivan** [comment]
@zachmullen, is this a bad idea to not limit the search results?

I could see this possibly blowing up but we need #354 resolved ASAP for a demo and current use cases do not have a ton of data

**Comment by zachmullen** [comment]
>  is this a bad idea to not limit the search results?

I think removing pagination altogether is the wrong solution for the problem as stated. For results sets that contain thousands of records, using paging is a very appropriate approach. What I would do is have the client library handle paging internally in a way that's transparent to the end user, perhaps with a generator as the API.

**Comment by matt-bernstein** [comment]
After realizing what `offset` was for, I was able to make `rgdc.Rgdc.search` behave correctly with the following snippet:

```python
    def _search(instrumentation):
        offset = 0
        results = []
        while (query := client.search(**kwargs, instrumentation=instrumentation, offset=offset)):
            # fix dates
            query = ([
                entry for entry in query if (dt_min <= datetime.fromisoformat(
                    entry['acquisition_date'].strip('Z')) <= dt_max)
            ])
            # fix dupes
            _, ixs = np.unique([entry['detail'] for entry in query],
                               return_index=True)
            query = list(np.array(query)[ixs])

            results.extend(query)

            offset += kwargs['limit']

        return results
```
and I agree that pagination is useful for these very large result sets (which will cease to be very large when the other issues in #354 are fixed, but other use cases may continue to be).

I'd support something like having `rgdc.Rgdc.search` do this automatically when `limit=None`.

**Comment by banesullivan** [comment]
Closing this in favor of #360 and #362

**Pull request closed without merging**